libs/data-mapper/src/lib/components/inputTypes/InputTextbox.tsx

```diff
@@ -0,0 +1,91 @@
+import { setConnectionInput } from '../../core/state/DataMapSlice';
+import type { RootState } from '../../core/state/Store';
+import type { FunctionData, FunctionInput } from '../../models';
+import { LogCategory, LogService } from '../../utils/Logging.Utils';
+import { Tooltip, Textarea } from '@fluentui/react-components';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+export type InputTextboxProps = {
+  input: FunctionInput;
+  functionNode: FunctionData;
+  loadedInputValue: string | undefined;
+};
+
+export const InputTextbox = ({ input, functionNode, loadedInputValue }: InputTextboxProps) => {
+  const selectedItemKey = useSelector((state: RootState) => state.dataMap.curDataMapOperation.selectedItemKey);
+  const [inputTyped, setInputTyped] = useState<boolean>(false);
+
+  const [inputText, setInputText] = useState<string>(loadedInputValue ? loadedInputValue : '');
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (loadedInputValue !== undefined && inputText === '' && inputTyped === false) {
+      const formattedValue = removeQuotesFromString(loadedInputValue);
+      setInputText(formattedValue);
+    }
+  }, [loadedInputValue, inputText, inputTyped]);
+
+  useEffect(() => {
+    const onCustomTextBoxChange = (value: string) => {
+      setInputTyped(true);
+      if (!selectedItemKey) {
+        LogService.error(LogCategory.InputTextbox, 'updateInput', {
+          message: 'Attempted to update input with nothing selected on canvas',
+        });
+
+        return;
+      }
+
+      const formattedValue = addQuotesToString(value);
+
+      if (inputTyped) {
+        dispatch(
+          setConnectionInput({
+            targetNode: functionNode,
+            targetNodeReactFlowKey: selectedItemKey,
+            inputIndex: 0,
+            input: formattedValue,
+          })
+        );
+      }
+    };
+    const timeOutId = setTimeout(() => onCustomTextBoxChange(inputText), 500);
```

Comment:
Is this just to debounce? If so take a look at https://github.com/react-hookz/web if you haven't. Might be some options there. We use some of the features from this elsewhere

Reply:
done